protocol/messenger_communities.go

```diff
@@ -1854,10 +1854,28 @@ func (m *Messenger) findCommunityInfoFromDB(communityID string) (*communities.Co
 	return community, nil
 }
 
+func getCommunityIdFromKey(communityKey string) (string, error) {
+	// Check if the key is a private key
+	privateKey, err := crypto.HexToECDSA(communityKey[2:])
+	communityID := ""
+	if err != nil {
+		// Not a private key, use the public key
+		communityID = communityKey
```

Comment:
Shouldn't we check if pubkey is valid here?

Reply:
I also think this should be checked. Although, I do see we haven't made such a check before either. This would ultimately fail when it doesn't find a community given that (invalid) key.